kustomize/commands/edit/set/setlabel.go

```diff
@@ -17,6 +18,7 @@ import (
 type setLabelOptions struct {
 	metadata     map[string]string
 	mapValidator func(map[string]string) error
+	fields       string // TODO does enum type exists in Go?
```

Comment:
Usually what I'll see people do is just do something like 

```
type Field string

const (
    commonLabels Field = "commonLabels" 
}

This might be a bit overkill for this PR so I think what you've done is fine.